### DISCUSSION TITLE
Only some classes working in Next.js

### DISCUSSION THREAD

**Discussion by taylorhlabs** [tailwindlabs/tailwindcss#3266] [Help] [answered]
I just added Tailwind to an existing Next.js 10 project using the Next.js setup guide [here](https://tailwindcss.com/docs/guides/nextjs). I ran the following commands per the guide:
```
npm install tailwindcss@latest postcss@latest autoprefixer@latest
```
```
npx tailwindcss init -p
```
I confirmed config files had been created then followed directions to "Import Tailwind directly in your JS" by updating the _app.js file. Note that the next-auth starter project I'm using uses the "styles.css" to add some very basic styling. (I tried commenting it out but it didn't help the issue.)
```
import './styles.css';
import 'tailwindcss/tailwind.css';
```
Next, I went into my project and started adding classes to some divs. For instance, I added the code below and then ran my local dev environment. 
```
<div key={resource.id} className="my-2 py-2 divide-y divide-gray-100">
```
For some reason, only the margin and padding classes are being applied. The divider classes are not working. In the Chrome Inspector, the divider classes do not even show up in the Styles pane; only the margin and padding classes do. It's like the divider classes don't exist.

I also tried running the default config command below to see if it helped, but I still have the same issue.
```
npx tailwind init tailwind.js --full
``` 
Do I need to do additional configuration to get the divider styles working? My project uses some "[name].module.css" files for styling Header and Footer components used by the global Layout component. Do these cause conflicts? No idea why only a few classes are working. Thanks in advance for any help or tips!

**Comment by taylorhlabs** [comment, marked as answer]
After looking at this again, it seems I was misusing the "divider" classes which need to be used on parent elements, not the children you want to have the divider applied to. 🙄

**Reply by simonswiss** [reply]
Yep - `divide` utilities (as well as `space-x` and `space-y` utilities) are to be used on the parent element!

Glad you got this sorted 👍

**Reply by shanitripathi** [reply]
The same issue is happening with me, I am trying to use top-1/2 and it doesnt work on the child element but when I use top-0, it works. top-0 is also a divider class, dont know what the issue is there

**Comment by Seanmclem** [comment]
I had a similar issue and found this discussion while searching. My solution was that I had to populate some _variants_, like from [this doc](https://tailwindcss.com/docs/configuring-variants#default-variants-reference). I just copied all the variants. Before, I just had an empty `variants` in my `tailwind.config.js` and I guess without some added variants, lots of styles won't work. This happened in nextjs, maybe because I copied my config from somewhere.

**Reply by MaximeTroiano** [reply]
Thanks for this answer, it's a dirty quick fix that helps a lot 😎

**Reply by connorlindsey** [reply]
Thanks! That fixed an issue I was having where tailwind classes I was adding were on the html element but not applying styles, super frustrating but this seems to works. 

Maybe a bug or at least warrants something in the docs

**Reply by Seanmclem** [reply]
@MaximeTroiano @connorlindsey after spending another day with this I feel like my fix is not actually that fixed. Seems like a workaround that works temporarily or intermittently, is to make an update, any update to your config file and save it. The rebuilding afterwards will include the Styles you just added. The jit compilation or somehow related to webpack 5, causes it to not include newly added styles when you just save.

**Reply by jsjoeio** [reply]
Thank you for posting your solution! I followed the official Next.js guide in the Tailwind docs and realized background colors (and other properties) were not working. Adding a field to `variants.extend` in my `tailwind.config.js` worked.

**Reply by jsjoeio** [reply]
Oddly enough, I can't reproduce it with a fresh Next.js + Tailwind app. The mystery continues.

**Reply by Seanmclem** [reply]
> Oddly enough, I can't reproduce it with a fresh Next.js + Tailwind app. The mystery continues. 

I think updating to the latest version(s) might fix the issue

**Reply by moschiel** [reply]
It worked!

**Comment by jtontiwith** [comment]
`module.exports = {
  content: [
    "./src/**/*.{js,ts,jsx,tsx}",
    // Add extra paths here
  ],
  theme: {
    extend: {},
  },
  plugins: [],
}`

From: https://stackoverflow.com/questions/66484296/some-tailwind-styles-not-working-in-production-with-next-js

**Reply by jsjoeio** [reply]
A little easier to read:

```js
module.exports = { 
  content: 
    [ "./src/**/*.{js,ts,jsx,tsx}", 
    // Add extra paths here 
    ], 
  theme: { 
    extend: {}, 
  }, 
  plugins: [], 
}
